streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java

```diff
@@ -422,4 +321,91 @@ private boolean isYarnNotDefaultQueue(FlinkCluster cluster) {
+            flinkEnvService.getById(flinkCluster.getVersionId()).getFlinkVersion(),
+            flinkCluster.getExecutionModeEnum(),
+            flinkCluster.getProperties(),
+            clusterId,
+            getExtendDeployDesc(flinkCluster, "shutdown"));
+    Future<ShutDownResponse> future =
+        executorService.submit(() -> FlinkClient.shutdown(stopRequest));
+    return future.get(60, TimeUnit.SECONDS);
+  }
+
+  private DeployResponse startInternal(FlinkCluster flinkCluster)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    DeployRequest deployRequest =
+        new DeployRequest(
+            flinkEnvService.getById(flinkCluster.getVersionId()).getFlinkVersion(),
+            flinkCluster.getExecutionModeEnum(),
+            flinkCluster.getProperties(),
+            flinkCluster.getClusterId(),
+            getExtendDeployDesc(flinkCluster, "start"));
+    log.info("Deploy cluster request " + deployRequest);
+    Future<DeployResponse> future = executorService.submit(() -> FlinkClient.deploy(deployRequest));
+    return future.get(60, TimeUnit.SECONDS);
+  }
+
+  private void checkActiveIfNeeded(FlinkCluster flinkCluster) {
+    if (ExecutionMode.YARN_SESSION.equals(flinkCluster.getExecutionModeEnum())) {
+      ApiAlertException.throwIfFalse(
+          ClusterState.STARTED.getValue().equals((flinkCluster.getClusterState())),
+          "Current cluster is not active, please check!");
+      if (!flinkCluster.verifyClusterConnection()) {
+        flinkCluster.setAddress(null);
+        flinkCluster.setClusterState(ClusterState.LOST.getValue());
+        updateById(flinkCluster);
+        throw new ApiAlertException("Current cluster is not active, please check!");
+      }
+    }
+  }
+
+  private void updateCluster(FlinkCluster cluster, FlinkCluster flinkCluster) {
+    flinkCluster.setClusterName(cluster.getClusterName());
+    flinkCluster.setDescription(cluster.getDescription());
+    if (ExecutionMode.REMOTE.equals(flinkCluster.getExecutionModeEnum())) {
+      flinkCluster.setAddress(cluster.getAddress());
+    } else {
+      flinkCluster.setAddress(null);
+      flinkCluster.setClusterId(cluster.getClusterId());
+      flinkCluster.setVersionId(cluster.getVersionId());
+      flinkCluster.setDynamicProperties(cluster.getDynamicProperties());
+      flinkCluster.setOptions(cluster.getOptions());
+      flinkCluster.setResolveOrder(cluster.getResolveOrder());
+      flinkCluster.setK8sHadoopIntegration(cluster.getK8sHadoopIntegration());
+      flinkCluster.setK8sConf(cluster.getK8sConf());
+      flinkCluster.setK8sNamespace(cluster.getK8sNamespace());
+      flinkCluster.setK8sRestExposedType(cluster.getK8sRestExposedType());
+      flinkCluster.setServiceAccount(cluster.getServiceAccount());
+      flinkCluster.setFlinkImage(cluster.getFlinkImage());
+      flinkCluster.setYarnQueue(cluster.getYarnQueue());
+    }
+  }
+
+  @Nullable
+  private KubernetesDeployParam getExtendDeployDesc(
```

Comment:
Since this method is only used for get deploy params for kubernetes mode, maybe we can name it `getKubernetesDeployDesc` ?